tests/integration/base.py

```diff
@@ -0,0 +1,327 @@
+            for event_file in event_files:
+                if self.function_id_by_event:
+                    cmdlist = [
+                        "sam",
+                        "local",
+                        "invoke",
+                        self.function_id_by_event[event_file],
+                        "-e",
+                        Path("events", event_file),
+                    ]
+                else:
+                    cmdlist = [
+                        "sam",
+                        "local",
+                        "invoke",
+                        "-e",
+                        Path("events", event_file),
+                    ]
+                LOG.info(cmdlist)
+                result = run_command(cmdlist, self.cwd)
+                try:
+                    self.invoke_output = json.loads(result.stdout)
+                except json.decoder.JSONDecodeError:
+                    self.fail(f"Response is not a valid JSON: {result.stdout}")
+
+        @pytest.mark.flaky(reruns=3)
+        def test_buld_and_invoke(self):
+            self._test_init_template()
+            self._test_build()
+            self._test_local_invoke()
+
+    class HelloWorldWithLocationBuildInvokeBase(BuildInvokeBase):
+        def _test_local_invoke(self):
+            super()._test_local_invoke()
+            self.assertEqual(self.invoke_output["statusCode"], 200)
+            self.assertEqual(
+                self.invoke_output["headers"],
+                {"X-Custom-Header": "application/json", "Content-Type": "application/json",},
+            )
+            body = json.loads(self.invoke_output["body"])
+            self.assertEqual(body["message"], "hello world")
+            # make sure it is an IP address
+            try:
+                ipaddress.ip_address(body["location"])
+            except ValueError:
+                self.fail(f'Invalid location: {body["location"]}')
+
+    class EventBridgeHelloWorldBuildInvokeBase(BuildInvokeBase):
+        def _test_local_invoke(self):
+            super()._test_local_invoke()
+            self.assertEqual(
+                self.invoke_output["detail"], {"instance-id": "i-abcd1111", "state": "pending"},
+            )
+            self.assertEqual(
+                self.invoke_output["detail-type"],
+                "HelloWorldFunction updated event of EC2 Instance State-change Notification",
+            )
+            self.assertEqual(
+                self.invoke_output["resources"], ["arn:aws:ec2:us-east-1:123456789012:instance/i-abcd1111"],
+            )
+            self.assertEqual(self.invoke_output["source"], "aws.ec2")
+            self.assertEqual(self.invoke_output["account"], "123456789012")
+            self.assertEqual(self.invoke_output["region"], "us-east-1")
+
+    class HelloWorldExclamationBuildInvokeBase(BuildInvokeBase):
+        def _test_local_invoke(self):
+            super()._test_local_invoke()
+            self.assertEqual(self.invoke_output["statusCode"], 200)
```

Comment:
not major, but most of 200 checks can be moved to method in parent.

Reply:
Not all `local invoke` response has `statusCode` in it. For example:

"java8.al2/cookiecutter-aws-sam-eventbridge-hello-java-gradle" 's response:

```
{"detail":{"instance-id":"i-abcd1111","state":"pending"},"detail-type":"HelloWorldFunction updated event of EC2 Instance State-change Notification","resources":["arn:aws:ec2:us-east-1:123456789012:instance/i-abcd1111"],"id":"7bf73129-1428-4cd3-a780-95db273d1602","source":"aws.ec2","time":1447277394000,"region":"us-east-1","version":null,"account":"123456789012"}
```